fetch/handler.go

```diff
@@ -59,6 +63,21 @@ func (h *handler) handleMaliciousIDsReq(ctx context.Context, _ []byte) ([]byte,
 	return data, nil
 }
 
+func (h *handler) handleMaliciousIDsReqStream(ctx context.Context, msg []byte, s io.ReadWriter) error {
+	if err := h.streamIDs(ctx, s, func(cbk retrieveCallback) error {
+		return identities.IterateMalicious(h.cdb, func(total int, id types.NodeID) error {
```

Comment:
i don't think that this is safe if we are using same connection pool for consensus and fetcher. this may block on the write to the stream

Reply:
Will do in a followup. Also probably we should support SQL query cache (which is LRU now so usable for end-users) not just for activesets and ATXs; with query cache enabled, `Iterate...` funcs use cached slice under the hood instead of holding the database connection for long. This way, we'll avoid blocking the database connections but also reduce memory use